controllers/front/Payment.php

```diff
@@ -514,6 +509,26 @@ private function processLocalPaymentMethod(\Cart $cart, $paymentType, $paymentIs
                     // invalid parameters
                     $isValid = false;
                 }
+                break;
+            case 'PresentToShopper':
+
+                $_SESSION['paymentAction'] = $response['action'];
+
+                $customer = new \Customer($cart->id_customer);
```

Comment:
should we not validate if $cart->id_customer has a value before loading it into the customer object ?

Reply:
prestashop will add an id_customer even if it is a guest user. If by any chance there is an issue with the customer id the next line: Validate::isLoadedObject($customer) will validate it.